pkg/inject/inject.go

```diff
@@ -629,7 +627,8 @@ func (conf *ResourceConfig) proxyResourceRequirements() v1.ResourceRequirements
 }
 
 func (conf *ResourceConfig) proxyDestinationAddr() string {
-	dns := fmt.Sprintf("linkerd-destination.%s.svc.cluster.local", conf.globalConfig.GetLinkerdNamespace())
+	ns := conf.configs.GetGlobal().GetLinkerdNamespace()
+	dns := fmt.Sprintf("linkerd-destination.%s.svc.cluster.local", ns)
```

Comment:
Shouldn't `cluster.local` be replaced with the `identity-trust-domain` option, or are these two separate things?

Reply:
The trust domain is explicitly decoupled from DNS: we continue to connect to "svc.cluster.local" DNS names (until we accept cluster name configurations later) -- but we validate peer identity based on the identity name with the trust-domain suffix. https://github.com/linkerd/linkerd2/pull/2536 introduces peer identity configuration.